src/basic/unit-name.c

```diff
@@ -507,6 +514,68 @@ int unit_name_template(const char *f, char **ret) {
         return 0;
 }
 
+bool unit_name_is_hashed(const char *name) {
+        char *s;
+
+        if (!unit_name_is_valid(name, UNIT_NAME_PLAIN))
+                return false;
+
+        s = strrchr(name, '.');
+
+        if (s - name < UNIT_NAME_HASH_LENGTH_CHARS + 1)
+                return false;
+
+        s -= UNIT_NAME_HASH_LENGTH_CHARS + 1;
+        if (*s++ != '_')
```

Comment:
hmm, first decrease then right after increase in a very cryptic way? why not do it like this:

```
s -= UNIT_NAME_HASH_LENGTH_CHARS;
if (s[-1] != '_')
        return false;
```

much less cryptic, at least in my eyes, no?